src/libstd/sync/mpsc/sync.rs

```diff
@@ -257,6 +280,26 @@ impl<T> Packet<T> {
         Ok(ret)
     }
 
+    pub fn recv_timeout(&self, timeout: Duration) -> Result<T, Failure> {
```

Comment:
Can this be refactored with `recv` above to share code? Right now it looks like they were basically just copy/pasted functions.